### PR TITLE
coverage content-length to string

### PR DESCRIPTION
**Pull request by makeco**
this pr show the problem in #465
> the typeof res.headers['content-length'] is not string, it is actually a number.

I have debug in `light-my-request` for a long time, at lastest I found it in `fastify/lib/reply.js`.

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ function onSendEnd (reply, payload) {
   }
 
   if (!reply.res.getHeader('Content-Length')) {
-    reply.res.setHeader('Content-Length', Buffer.byteLength(payload))
+    reply.res.setHeader('Content-Length', Buffer.byteLength(payload).toString())
```
Comment (allevo) [inline lib/reply.js]: `'' + Buffer.byteLength(payload)`

Comment (jsumners) [reply lib/reply.js]: @allevo why? `.toString` seems clearer. Is there a performance difference?

Comment (mcollina) [reply lib/reply.js]: I haven't benchmarked it, but I suspect there is. not sure it's noticeable.

Comment (delvedor) [reply lib/reply.js]: `'' + Buffer.byteLength(payload)` is faster.

**Comment by makeco** [comment]
I think `light-my-request` test file need some patch. And I am honored to contribute.

**Comment by delvedor** [comment]
I was reading the [HTTP/1.1 RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) specification, who says:

> The Content-Length entity-header field indicates the size of the entity-body, in decimal number of OCTETs, sent to the recipient or, in the case of the HEAD method, the size of the entity-body that would have been sent had the request been a GET.

Correct me if I'm wrong, but the `content-length` should be a number.
cc @mcollina @jsumners @allevo

**Comment by jsumners** [comment]
Well, a string representation of a number. All headers are strings.

**Comment by delvedor** [comment]
@jsumners you are right, I've just checked also the node docs.
Thanks!

**Pull request merged**